File: src/main/java/fi/finlit/edith/sql/domain/Place.java
package fi.finlit.edith.sql.domain;

import java.util.HashSet;
import java.util.Set;

import javax.persistence.CollectionTable;
import javax.persistence.ElementCollection;
import javax.persistence.Embedded;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.Table;

@Entity
@Table(name = "place")
public class Place {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;

    @Embedded
    private NameForm normalized;

    @ElementCollection
    @CollectionTable(name = "place_nameform", joinColumns = @JoinColumn(name = "place_id"))
    private Set<NameForm> otherForms = new HashSet<NameForm>();

    public Place() {
    }

    public Place(NameForm normalized, Set<NameForm> otherForms) {
        this.normalized = normalized;
        this.otherForms = otherForms;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public NameForm getNormalized() {
        return normalized;
    }

    public Set<NameForm> getOtherForms() {
        return otherForms;
    }

    public void setNormalized(NameForm normalized) {
        this.normalized = normalized;
    }

    public void setOtherForms(Set<NameForm> otherForms) {
        this.otherForms = otherForms;
    }

}

File: src/main/scala/fi/finlit/edith/sql/domain/Place.scala
package fi.finlit.edith.sql.domain

import java.util.HashSet
import java.util.Set
import javax.persistence.CollectionTable
import javax.persistence.ElementCollection
import javax.persistence.Embedded
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.JoinColumn
import javax.persistence.Table
import fi.finlit.edith.Identifiable
import scala.reflect.BeanProperty
//remove if not needed
import scala.collection.JavaConversions._

@Entity
@Table(name = "place")
class Place() extends Identifiable {

  @Id
  @GeneratedValue(strategy = GenerationType.AUTO)
  @BeanProperty
  var id: java.lang.Long = _

  @Embedded
  @BeanProperty
  var normalized: NameForm = _

  @ElementCollection
  @CollectionTable(name = "place_nameform", joinColumns = @JoinColumn(name = "place_id"))
  @BeanProperty
  var otherForms = new HashSet[NameForm]()

  def this(normalized: NameForm, otherForms: Set[NameForm]) {
    this.normalized = normalized
    this.otherForms = otherForms
  }
}
